### PR TITLE
feat: ws1 1455 e2e test for get events

### PR DESCRIPTION
**Pull request by samika98**
E2E tests for fetching event data from event_id (string). 
Goal is to test the getEvent endpoint end-to-end.
Scenarios tested : 
Positive scenario: StatusCode 200 : Publish and event, and then fetch event data for the same event
Negative scenario: StatusCode 404 (Event Not found) : Fetch eventData for an eventId that does not exist

PR testing : 
Ran the setup locally : 
Screenshot 2024-02-06 at 3 58 01 PM

### CODE REVIEW

File: suite/src/__tests__/fast/get-events.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  test('get event data for non-existing event', async () => {
+    const eventId = generateRandomEventId()
+    // fetching the event from its event-id from rust-ceramic
+    const getResponse = await getEventData(ceramicUrl, eventId)
+    const responseText = await getResponse.text()
+    expect(getResponse.status).toEqual(400)
+    expect(responseText).toContain('Event not found')
+    expect(responseText).toContain(eventId)
+    expect(getResponse.statusText).toContain('Bad Request')
```
Comment (dav1do) [inline suite/src/__tests__/fast/get-events.test.ts:50]: I don't think you need to check for the "bad request" text here. Also, as I commented on the PR, I think we should return 404 for not found, and 400 if the event Id is invalid (but we don't have to do all of that yet). Anyway, if you do change that status code this will change too.

File: suite/src/utils/rustCeramicHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { randomCID, EventID, StreamID } from '@ceramicnetwork/streamid'
+import { base16 } from 'multiformats/bases/base16'
+import { CARFactory } from 'cartonne'
+import { base64 } from 'multiformats/bases/base64'
+import * as random from '@stablelib/random'
+
+export function generateRandomEventId(): string {
```
Comment (dav1do) [inline suite/src/utils/rustCeramicHelpers.ts:7]: maybe want to update the `fast/events.ts` random event generation to use these functions too?

**Review by dav1do — COMMENTED** [review]
Looking good! Some minor comments. I'd like to merge your changes in rust-ceramic and then make sure these past, the 404s make sense right now since the new code isn't deployed.

File: suite/src/utils/rustCeramicHelpers.ts (hunk 2 above)
Comment (samika98) [reply suite/src/utils/rustCeramicHelpers.ts:7]: Good catch! Don't think it should be in this PR though cause it does not have anything to do with the e2e test for fetching events

File: suite/src/__tests__/fast/get-events.test.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  test('post and get event data, success', async () => {
+    const eventId = generateRandomEventId()
+    const event = generateRandomEvent(eventId)
+    // publishing the event to rust-ceramic
+    await postEvent(ceramicUrl, event)
+    // fetching the event from its event-id from rust-ceramic
+    const getResponse = await getEventData(ceramicUrl, eventId)
+    expect(getResponse.status).toEqual(200)
+    expect(await getResponse.json()).toEqual({
+      eventId: eventId,
```
Comment (dav1do) [inline suite/src/__tests__/fast/get-events.test.ts:37]: Assuming my [changes land](https://github.com/ceramicnetwork/rust-ceramic/pull/255), I modified the event response to be `{id, data}` to save some bytes based on review. This only applies to responses (so the `GET event/$id` and `GET feed/events` endpoints). The input is still `eventId/eventData` to avoid breaking existing functionality.

File: suite/src/utils/rustCeramicHelpers.ts (hunk 2 above)
Comment (smrz2001) [reply suite/src/utils/rustCeramicHelpers.ts:7]: Would just be a minor refactor though? Shouldn't add too much to this PR's scope and the code comes out cleaner versus someone going in afterwards in a separate PR just for reducing code duplication. WDYT?

I don't have a strong opinion though, it's cool to open a ticket for cleanup, maybe tagged as `debt`.

Deciding whether to change adjacent code in a PR is more of an art than a science. In PRs with a lot of moving parts, unrelated changes can certainly be a distraction.

Comment (samika98) [reply suite/src/utils/rustCeramicHelpers.ts:7]: Created a separate issue for the clean-up. I am planning to move all re-usable functions (not too many) from event.rs to the utils. 
Can save us some time in the future :)
https://linear.app/3boxlabs/issue/WS1-1475/clear-some-technical-debt-for-rust-ceramic-and-cermic-tests

File: suite/src/__tests__/fast/get-events.test.ts (hunk 1 above)
Comment (samika98) [reply suite/src/__tests__/fast/get-events.test.ts]: Updated the code

File: suite/src/utils/rustCeramicHelpers.ts (hunk 2 above)
Comment (dav1do) [reply suite/src/utils/rustCeramicHelpers.ts:7]: Thanks, ticket looks good. I agree with Mohsin that there is a lot of nuance to refactors. In this case, we're still adding and changing these tests, so I don't think it's a problem to follow up. 

My general thinking is this: as long as it doesn't blow up the testing scope, cause breaking changes, make review a lot harder, take way longer, or cause difficult merge conflicts, it's worth doing now. For the review piece, putting those refactoring changes (e.g. moving code across files or renaming) in their own commits can make it easier. 

Basically, my preference is to do any refactoring I can when I'm in the code because who knows when I'll be back and have the chance, and there's the overhead of reloading the context into my brain and getting another review etc. It's nice to whittle a little tech debt every PR if possible. I think it's harder to sell "our cycle is tackling tech debt" than "we're doing these features and these tickets are a few extra points so we can take care of things" to leadership.

**Review by dav1do — APPROVED** [review]
Looks good! 🚀

Mind updating the summary? It says 500 instead of 200 for successful test, and the picture upload got stuck, so you can try that again or just delete that piece.

File: suite/src/utils/rustCeramicHelpers.ts (hunk 2 above)
Comment (samika98) [reply suite/src/utils/rustCeramicHelpers.ts:7]: I agree, on addition it might be to remove debt on the file being touched in the PR, this might be a more standardized way of removing debt. Since a developer has visibility of the file being worked on, introducing refactoring to another file that is not of interest to the PR might increase scope in review rather than having a predefined scope for the task
